index.js

```diff
@@ -749,12 +746,22 @@ RedisClient.prototype.send_command = function (command, args, callback) {
         args = args.slice(0, -1).concat(args[args.length - 1]);
     }
 
-    // if the value is undefined or null and command is set or setx, need not to send message to redis
-    if (command === 'set' || command === 'setex') {
-        if(args[args.length - 1] === undefined || args[args.length - 1] === null) {
-            var err = new Error('send_command: ' + command + ' value must not be undefined or null');
-            return callback && callback(err);
-        }
+    // if the value or the key is undefined or null and command is set, setex or mset; need not to send message to redis
+    if (command === 'set' || command === 'setex' || command === 'mset') {
```

Comment:
We might consider the change as a bug fix though. Using `null` and `undefined` as a key for set is not reasonable. And there are a few issues where people complain that they overwrite old values by accident because they did not check that the value was undefined or null.
I decided to remove this part from the jshint PR and make an individual one to fix that issue including the therefor broken tests.